pkgs/build-support/fetchgit/default.nix

```diff
@@ -13,7 +13,10 @@ in
 lib.makeOverridable (lib.fetchers.withNormalizedHash { } (
 # NOTE Please document parameter additions or changes in
 #   doc/build-helpers/fetchers.chapter.md
-{ url, rev ? "HEAD", leaveDotGit ? deepClone
+{ url
+, tag ? null
+, rev ? if tag != null then "refs/tags/${tag}" else "HEAD" # FIXME fetching HEAD by default is problematic at best
```

Comment:
As a follow-on PR, what do you think about making the problematic default `lib.warn "fetching HEAD by default; this will become an error in xx.yy" "HEAD"` or something similar here?